{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "pd.options.display.max_rows = 4000\n",
    "pd.options.display.max_colwidth = 200\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def readData(path):\n",
    "    \n",
    "    with open(path, \"r\", encoding=\"utf-8\") as file:\n",
    "        data = json.load(file)\n",
    "\n",
    "    TEXT = []\n",
    "    ENTITY_NAME = []\n",
    "    ENTITY_LABEL = []\n",
    "#     IDS = []\n",
    "    for i, dataItem in enumerate(data):\n",
    "        entitiesDict = {}\n",
    "        entitiesDict[\"entities\"] = []\n",
    "        for slotItem in dataItem[\"slots\"]:\n",
    "            TEXT.append(dataItem[\"text\"])\n",
    "            ENTITY_NAME.append(slotItem[\"slotValue\"])\n",
    "            ENTITY_LABEL.append(slotItem[\"slotName\"])\n",
    "            \n",
    "    return pd.DataFrame({\"text\": TEXT, \"name\": ENTITY_NAME, \"label\": ENTITY_LABEL})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "PATH = \"/home/abbhatt/Documents/flairExperiments/data/train-refactored.json\"\n",
    "df_train = readData(PATH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_train[\"length\"] = df_train[\"name\"].apply(lambda x: len(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Moreover, the in vitro binding of NF-κB or Sp1 to its target DNA was not affected by the presence of K-12.</td>\n",
       "      <td>DNA</td>\n",
       "      <td>DNA</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Moreover, the in vitro binding of NF-κB or Sp1 to its target DNA was not affected by the presence of K-12.</td>\n",
       "      <td>Sp1</td>\n",
       "      <td>protein</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Moreover, the in vitro binding of NF-κB or Sp1 to its target DNA was not affected by the presence of K-12.</td>\n",
       "      <td>NF-κB</td>\n",
       "      <td>protein</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Moreover, the in vitro binding of NF-κB or Sp1 to its target DNA was not affected by the presence of K-12.</td>\n",
       "      <td>K-12</td>\n",
       "      <td>chemical</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>No appreciable binding of Ly49C to Db could be detected in the cell-cell adhesion assay by antibody blocking or by employing cells from Kb / mice that express Db normally ( Figure 3 and Figure 4).</td>\n",
       "      <td>Db</td>\n",
       "      <td>protein</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                   text  \\\n",
       "0                                                                                            Moreover, the in vitro binding of NF-κB or Sp1 to its target DNA was not affected by the presence of K-12.   \n",
       "1                                                                                            Moreover, the in vitro binding of NF-κB or Sp1 to its target DNA was not affected by the presence of K-12.   \n",
       "2                                                                                            Moreover, the in vitro binding of NF-κB or Sp1 to its target DNA was not affected by the presence of K-12.   \n",
       "3                                                                                            Moreover, the in vitro binding of NF-κB or Sp1 to its target DNA was not affected by the presence of K-12.   \n",
       "4  No appreciable binding of Ly49C to Db could be detected in the cell-cell adhesion assay by antibody blocking or by employing cells from Kb / mice that express Db normally ( Figure 3 and Figure 4).   \n",
       "\n",
       "    name     label  length  \n",
       "0    DNA       DNA       3  \n",
       "1    Sp1   protein       3  \n",
       "2  NF-κB   protein       5  \n",
       "3   K-12  chemical       4  \n",
       "4     Db   protein       2  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_train.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "PATH = \"/home/abbhatt/Documents/flairExperiments/data/dev-refactored.json\"\n",
    "df_dev = readData(PATH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_dev[\"length\"] = df_dev[\"name\"].apply(lambda x: len(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Taken together, these results make it clear that ATP-bound forms of ORC and Cdc6p are likely to be required for productive interactions and pre-RC formation.</td>\n",
       "      <td>Cdc6p</td>\n",
       "      <td>protein</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Taken together, these results make it clear that ATP-bound forms of ORC and Cdc6p are likely to be required for productive interactions and pre-RC formation.</td>\n",
       "      <td>ORC</td>\n",
       "      <td>protein-complex</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Taken together, these results make it clear that ATP-bound forms of ORC and Cdc6p are likely to be required for productive interactions and pre-RC formation.</td>\n",
       "      <td>ATP</td>\n",
       "      <td>chemical</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Taken together, these results make it clear that ATP-bound forms of ORC and Cdc6p are likely to be required for productive interactions and pre-RC formation.</td>\n",
       "      <td>pre-RC</td>\n",
       "      <td>protein-complex</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Additional experiments were therefore performed to clarify whether the Vps10p cd could in fact bind to human GGA1 and GGA2.</td>\n",
       "      <td>Vps10p</td>\n",
       "      <td>protein</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                            text  \\\n",
       "0  Taken together, these results make it clear that ATP-bound forms of ORC and Cdc6p are likely to be required for productive interactions and pre-RC formation.   \n",
       "1  Taken together, these results make it clear that ATP-bound forms of ORC and Cdc6p are likely to be required for productive interactions and pre-RC formation.   \n",
       "2  Taken together, these results make it clear that ATP-bound forms of ORC and Cdc6p are likely to be required for productive interactions and pre-RC formation.   \n",
       "3  Taken together, these results make it clear that ATP-bound forms of ORC and Cdc6p are likely to be required for productive interactions and pre-RC formation.   \n",
       "4                                    Additional experiments were therefore performed to clarify whether the Vps10p cd could in fact bind to human GGA1 and GGA2.   \n",
       "\n",
       "     name            label  length  \n",
       "0   Cdc6p          protein       5  \n",
       "1     ORC  protein-complex       3  \n",
       "2     ATP         chemical       3  \n",
       "3  pre-RC  protein-complex       6  \n",
       "4  Vps10p          protein       6  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_dev.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Checking the text of entities which are very less in number"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['DNA' 'RNA' 'RNA-family' 'amino-acid' 'assay' 'brand' 'cell' 'chemical'\n",
      " 'disease' 'drug' 'experiment-tag' 'experimental-construct'\n",
      " 'fusion-protein' 'gene' 'mutation' 'organelle' 'organism' 'parameter'\n",
      " 'peptide' 'process' 'protein' 'protein-DNA-complex' 'protein-RNA-complex'\n",
      " 'protein-complex' 'protein-domain' 'protein-family' 'protein-isoform'\n",
      " 'protein-motif' 'protein-region' 'reagent' 'tissue']\n",
      "31\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(None, None)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(np.unique(df_train[\"label\"])), print(len(np.unique(df_train[\"label\"])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "protein                   2595\n",
      "protein-family             752\n",
      "chemical                   389\n",
      "DNA                        363\n",
      "protein-complex            303\n",
      "protein-domain             235\n",
      "RNA                        110\n",
      "cell                       106\n",
      "experimental-construct     104\n",
      "experiment-tag              99\n",
      "gene                        98\n",
      "protein-motif               98\n",
      "reagent                     92\n",
      "amino-acid                  57\n",
      "protein-region              52\n",
      "assay                       38\n",
      "organelle                   30\n",
      "peptide                     26\n",
      "fusion-protein              24\n",
      "process                     20\n",
      "protein-isoform             20\n",
      "protein-RNA-complex         14\n",
      "mutation                    13\n",
      "drug                        11\n",
      "disease                      5\n",
      "organism                     4\n",
      "parameter                    4\n",
      "brand                        4\n",
      "protein-DNA-complex          3\n",
      "tissue                       2\n",
      "RNA-family                   2\n",
      "Name: label, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(df_train[\"label\"].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "protein                   348\n",
      "protein-family            118\n",
      "protein-complex            47\n",
      "chemical                   45\n",
      "DNA                        43\n",
      "protein-domain             34\n",
      "cell                       18\n",
      "reagent                    14\n",
      "experiment-tag             13\n",
      "protein-motif              11\n",
      "experimental-construct     10\n",
      "assay                       9\n",
      "organelle                   8\n",
      "peptide                     8\n",
      "protein-isoform             7\n",
      "RNA                         6\n",
      "drug                        5\n",
      "gene                        4\n",
      "protein-region              4\n",
      "process                     3\n",
      "amino-acid                  3\n",
      "protein-RNA-complex         3\n",
      "organism                    1\n",
      "fusion-protein              1\n",
      "disease                     1\n",
      "Name: label, dtype: int64\n",
      "25\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(None, None)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(df_dev[\"label\"].value_counts()), print(len(np.unique(df_dev[\"label\"])))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1) organism "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3207</th>\n",
       "      <td>Pseudomonas aeruginosa and Pseudomonas cepacia isolated from cystic fibrosis patients bind specifically to gangliotetraosylceramide ( Asialo GM1) and gangliotriaosylceramide ( Asialo GM2).</td>\n",
       "      <td>Pseudomonas cepacia</td>\n",
       "      <td>organism</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3208</th>\n",
       "      <td>Pseudomonas aeruginosa and Pseudomonas cepacia isolated from cystic fibrosis patients bind specifically to gangliotetraosylceramide ( Asialo GM1) and gangliotriaosylceramide ( Asialo GM2).</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>organism</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5576</th>\n",
       "      <td>Xenopus TCF3 binds to the distal Tcf-motif in the myf-5 gene in vitro.</td>\n",
       "      <td>Xenopus</td>\n",
       "      <td>organism</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5640</th>\n",
       "      <td>CK1epsilon stimulates the binding of dsh to GBP in Xenopus extracts.</td>\n",
       "      <td>Xenopus</td>\n",
       "      <td>organism</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                              text  \\\n",
       "3207  Pseudomonas aeruginosa and Pseudomonas cepacia isolated from cystic fibrosis patients bind specifically to gangliotetraosylceramide ( Asialo GM1) and gangliotriaosylceramide ( Asialo GM2).   \n",
       "3208  Pseudomonas aeruginosa and Pseudomonas cepacia isolated from cystic fibrosis patients bind specifically to gangliotetraosylceramide ( Asialo GM1) and gangliotriaosylceramide ( Asialo GM2).   \n",
       "5576                                                                                                                        Xenopus TCF3 binds to the distal Tcf-motif in the myf-5 gene in vitro.   \n",
       "5640                                                                                                                          CK1epsilon stimulates the binding of dsh to GBP in Xenopus extracts.   \n",
       "\n",
       "                        name     label  length  \n",
       "3207     Pseudomonas cepacia  organism      19  \n",
       "3208  Pseudomonas aeruginosa  organism      22  \n",
       "5576                 Xenopus  organism       7  \n",
       "5640                 Xenopus  organism       7  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_train[df_train[\"label\"] == \"organism\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>In Aplysia, as in mammals, the regulation of gene transcription by cAMP is mediated by cAMP Responsive Element ( CRE) binding protein CREB.</td>\n",
       "      <td>Aplysia</td>\n",
       "      <td>organism</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                            text  \\\n",
       "614  In Aplysia, as in mammals, the regulation of gene transcription by cAMP is mediated by cAMP Responsive Element ( CRE) binding protein CREB.   \n",
       "\n",
       "        name     label  length  \n",
       "614  Aplysia  organism       7  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_dev[df_dev[\"label\"] == \"organism\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1) Disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>570</th>\n",
       "      <td>In addition, changes to tau also regulated this interaction, as shown by the altered binding of S100β to the AD-related hyperphosphorylated NFT-tau.</td>\n",
       "      <td>AD</td>\n",
       "      <td>disease</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>792</th>\n",
       "      <td>The data presented in this study clearly show that I-DOX binds specifically to TTR amyloid fibrils in tissues from patients with FAP.</td>\n",
       "      <td>FAP</td>\n",
       "      <td>disease</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2160</th>\n",
       "      <td>To address the relative contributions of [3H]BTA-1 binding to Aβ and tau deposits in the frontal gray of AD brain we compared [3H]BTA-1 binding in homogenates from entorhinal cortex, frontal gray,...</td>\n",
       "      <td>AD</td>\n",
       "      <td>disease</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4081</th>\n",
       "      <td>( C) Binding of Sp1 to the D2 promoter is inhibited in the caudate of the human HD postmortem brain.</td>\n",
       "      <td>HD</td>\n",
       "      <td>disease</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4663</th>\n",
       "      <td>Because HSPGs have been detected immunohistochemically in the pancreatic amyloid deposits of type II diabetes ( 35), we investigated the binding of amylin to heparin.</td>\n",
       "      <td>type II diabetes</td>\n",
       "      <td>disease</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                         text  \\\n",
       "570                                                      In addition, changes to tau also regulated this interaction, as shown by the altered binding of S100β to the AD-related hyperphosphorylated NFT-tau.   \n",
       "792                                                                     The data presented in this study clearly show that I-DOX binds specifically to TTR amyloid fibrils in tissues from patients with FAP.   \n",
       "2160  To address the relative contributions of [3H]BTA-1 binding to Aβ and tau deposits in the frontal gray of AD brain we compared [3H]BTA-1 binding in homogenates from entorhinal cortex, frontal gray,...   \n",
       "4081                                                                                                     ( C) Binding of Sp1 to the D2 promoter is inhibited in the caudate of the human HD postmortem brain.   \n",
       "4663                                   Because HSPGs have been detected immunohistochemically in the pancreatic amyloid deposits of type II diabetes ( 35), we investigated the binding of amylin to heparin.   \n",
       "\n",
       "                  name    label  length  \n",
       "570                 AD  disease       2  \n",
       "792                FAP  disease       3  \n",
       "2160                AD  disease       2  \n",
       "4081                HD  disease       2  \n",
       "4663  type II diabetes  disease      16  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_train[df_train[\"label\"] == \"disease\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>402</th>\n",
       "      <td>Thus, is it that the Igs bind to an autoantigen(s) on specific neurons that ‘subsequently’ show signs of degeneration, to eventually express caspase 3 and die, due to the Ig binding (the AD autoim...</td>\n",
       "      <td>AD</td>\n",
       "      <td>disease</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                        text  \\\n",
       "402  Thus, is it that the Igs bind to an autoantigen(s) on specific neurons that ‘subsequently’ show signs of degeneration, to eventually express caspase 3 and die, due to the Ig binding (the AD autoim...   \n",
       "\n",
       "    name    label  length  \n",
       "402   AD  disease       2  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_dev[df_dev[\"label\"] == \"disease\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "3) drug"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1156</th>\n",
       "      <td>Moreover, the insulin-induced increase in GTP bound Rheb, but not that of Ras, is blocked by wortmannin, whereas both increases are resistant to rapamycin (Figure 2A).</td>\n",
       "      <td>rapamycin</td>\n",
       "      <td>drug</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1495</th>\n",
       "      <td>Additional evidence that FKBP12 specifically recognizes the Leu-Pro motif is provided by previous reports (Okadome et al., 1996; Wang et al., 1996b) and present observations that rapamycin and FK...</td>\n",
       "      <td>rapamycin</td>\n",
       "      <td>drug</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2481</th>\n",
       "      <td>When heat-shocked cells were also treated with rapamycin, an additive effect was observed such that rapamycin increased further both the amount of 4E-BP1 bound to eIF4E and the downward band shif...</td>\n",
       "      <td>rapamycin</td>\n",
       "      <td>drug</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2897</th>\n",
       "      <td>( GA) enhances the heat-inducible HSE-binding activity of HSF1 and delays deactivation during recovery.</td>\n",
       "      <td>( GA)</td>\n",
       "      <td>drug</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3127</th>\n",
       "      <td>Geldanamycin also prevents binding of the p23 cochaperone to HSP90 and inhibits the release of substrates ( 22, 85).</td>\n",
       "      <td>Geldanamycin</td>\n",
       "      <td>drug</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3153</th>\n",
       "      <td>In three experiments, the binding of Ras to Raf-1 peptide increased upon antigen stimulation and showed similar maximal increases at 1–2 min in both control (2.0 ± 0.5-fold, mean ± S.E.) and dexam...</td>\n",
       "      <td>dexamethasone</td>\n",
       "      <td>drug</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3463</th>\n",
       "      <td>( A) Effect of rapamycin on phosphorylation of eIF2alpha is strongly dependent on dephosphorylation of Ser 577 and the tRNA binding activity of GCN2.</td>\n",
       "      <td>rapamycin</td>\n",
       "      <td>drug</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4030</th>\n",
       "      <td>Thus, it appears that tRNA binding by GCN2 is absolutely required for its activation by rapamycin, as observed previously for amino acid limitation (Wek et al. 1995 ).</td>\n",
       "      <td>rapamycin</td>\n",
       "      <td>drug</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4356</th>\n",
       "      <td>In comparing DNA binding of Dm and MGTs, it should be noted that the substitution of the amino-terminal formyl and the carboxyl-terminal amidine groups of Dm by acetamido and -CH2CH2N+H(CH3)2 grou...</td>\n",
       "      <td>Dm</td>\n",
       "      <td>drug</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4358</th>\n",
       "      <td>In comparing DNA binding of Dm and MGTs, it should be noted that the substitution of the amino-terminal formyl and the carboxyl-terminal amidine groups of Dm by acetamido and -CH2CH2N+H(CH3)2 grou...</td>\n",
       "      <td>MGTs</td>\n",
       "      <td>drug</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4473</th>\n",
       "      <td>This time course of ATP inhibition correlates well with the activation of HSF binding by salicylate indicating that decreases in intracellular ATP may be an important factor in the activation of H...</td>\n",
       "      <td>salicylate</td>\n",
       "      <td>drug</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                         text  \\\n",
       "1156                                 Moreover, the insulin-induced increase in GTP bound Rheb, but not that of Ras, is blocked by wortmannin, whereas both increases are resistant to rapamycin (Figure 2A).    \n",
       "1495   Additional evidence that FKBP12 specifically recognizes the Leu-Pro motif is provided by previous reports (Okadome et al., 1996; Wang et al., 1996b) and present observations that rapamycin and FK...   \n",
       "2481   When heat-shocked cells were also treated with rapamycin, an additive effect was observed such that rapamycin increased further both the amount of 4E-BP1 bound to eIF4E and the downward band shif...   \n",
       "2897                                                                                                  ( GA) enhances the heat-inducible HSE-binding activity of HSF1 and delays deactivation during recovery.   \n",
       "3127                                                                                     Geldanamycin also prevents binding of the p23 cochaperone to HSP90 and inhibits the release of substrates ( 22, 85).   \n",
       "3153  In three experiments, the binding of Ras to Raf-1 peptide increased upon antigen stimulation and showed similar maximal increases at 1–2 min in both control (2.0 ± 0.5-fold, mean ± S.E.) and dexam...   \n",
       "3463                                                    ( A) Effect of rapamycin on phosphorylation of eIF2alpha is strongly dependent on dephosphorylation of Ser 577 and the tRNA binding activity of GCN2.   \n",
       "4030                                  Thus, it appears that tRNA binding by GCN2 is absolutely required for its activation by rapamycin, as observed previously for amino acid limitation (Wek et al. 1995 ).   \n",
       "4356  In comparing DNA binding of Dm and MGTs, it should be noted that the substitution of the amino-terminal formyl and the carboxyl-terminal amidine groups of Dm by acetamido and -CH2CH2N+H(CH3)2 grou...   \n",
       "4358  In comparing DNA binding of Dm and MGTs, it should be noted that the substitution of the amino-terminal formyl and the carboxyl-terminal amidine groups of Dm by acetamido and -CH2CH2N+H(CH3)2 grou...   \n",
       "4473  This time course of ATP inhibition correlates well with the activation of HSF binding by salicylate indicating that decreases in intracellular ATP may be an important factor in the activation of H...   \n",
       "\n",
       "               name label  length  \n",
       "1156      rapamycin  drug       9  \n",
       "1495      rapamycin  drug       9  \n",
       "2481      rapamycin  drug       9  \n",
       "2897          ( GA)  drug       5  \n",
       "3127   Geldanamycin  drug      12  \n",
       "3153  dexamethasone  drug      13  \n",
       "3463      rapamycin  drug       9  \n",
       "4030      rapamycin  drug       9  \n",
       "4356             Dm  drug       2  \n",
       "4358           MGTs  drug       4  \n",
       "4473     salicylate  drug      10  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_train[df_train[\"label\"] == \"drug\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>The effect was specific to rapamycin, as FK506, an immunosuppressant that also binds FKBP12 but does not target mTOR, had no effect on the interaction, nor did ethanol, the vehicle used for both d...</td>\n",
       "      <td>rapamycin</td>\n",
       "      <td>drug</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>As expected, the affinity of [3H]LY341495 in rat brain shown here is much higher than that observed for group II binding with [3H]glutamate (Kd= 187 nM).</td>\n",
       "      <td>[3H]LY341495</td>\n",
       "      <td>drug</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>Moreover, concentrations of tamoxifen as low as 0.25 muM substantially increased the binding of calmodulin to the estrogen receptor.</td>\n",
       "      <td>tamoxifen</td>\n",
       "      <td>drug</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>A shows the amino acid comparison of the three FK506 binding domains of dFKBP59 with human FKBP12 and FKBP12.6.</td>\n",
       "      <td>FK506</td>\n",
       "      <td>drug</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>GST beads bound to a fusion protein between GST and TIF2.1 or to GST alone were incubated with in vitro translated wild-type or mutant ERs in the presence of vehicle, estradiol, or antiestrogens (...</td>\n",
       "      <td>antiestrogens</td>\n",
       "      <td>drug</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                        text  \\\n",
       "115  The effect was specific to rapamycin, as FK506, an immunosuppressant that also binds FKBP12 but does not target mTOR, had no effect on the interaction, nor did ethanol, the vehicle used for both d...   \n",
       "190                                                As expected, the affinity of [3H]LY341495 in rat brain shown here is much higher than that observed for group II binding with [3H]glutamate (Kd= 187 nM).   \n",
       "191                                                                    Moreover, concentrations of tamoxifen as low as 0.25 muM substantially increased the binding of calmodulin to the estrogen receptor.    \n",
       "261                                                                                          A shows the amino acid comparison of the three FK506 binding domains of dFKBP59 with human FKBP12 and FKBP12.6.   \n",
       "456  GST beads bound to a fusion protein between GST and TIF2.1 or to GST alone were incubated with in vitro translated wild-type or mutant ERs in the presence of vehicle, estradiol, or antiestrogens (...   \n",
       "\n",
       "              name label  length  \n",
       "115      rapamycin  drug       9  \n",
       "190   [3H]LY341495  drug      12  \n",
       "191      tamoxifen  drug       9  \n",
       "261          FK506  drug       5  \n",
       "456  antiestrogens  drug      13  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_dev[df_dev[\"label\"] == \"drug\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>The yeast two-hybrid interaction assays indicated that filamin-A does not bind to any integrin β subunit, whereas filamin-B only interacts with β1A.</td>\n",
       "      <td>β1A</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>Experimental studies of liver regeneration, the acute-phase response to inflammation, and the pathology of myotonic dystrophy (DM) showed that, in all of these situations, CUGBP1 binds to the GCN ...</td>\n",
       "      <td>LIP</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1447</th>\n",
       "      <td>A neurobeachin isoform that does not bind RII, beige-like protein (BGL), is expressed in many tissues.</td>\n",
       "      <td>neurobeachin isoform</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1715</th>\n",
       "      <td>Dynamin 2, a binding partner of the SH3 domain of amphiphysin (9), was recruited to the tubules when coexpressed with M-amphiphysin 2 (Fig. 2D), but not when coexpressed with its BAR* domain, whic...</td>\n",
       "      <td>M-amphiphysin 2</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1991</th>\n",
       "      <td>Baker et al. (Baker et al., 2001 ) show in vitro experiments in which MAE binding to PNT-P2 leads to activation of transcription, and assume that this occurs via MAE promoting MAPK phosphorylatio...</td>\n",
       "      <td>PNT-P2</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2007</th>\n",
       "      <td>To test this hypothesis, we have examined the interaction of Arp1 with the Golgi-associated betaIII isoform of spectrin and demonstrated that Arp1 binds directly to betaIII spectrin.</td>\n",
       "      <td>Golgi-associated betaIII</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2008</th>\n",
       "      <td>To test this hypothesis, we have examined the interaction of Arp1 with the Golgi-associated betaIII isoform of spectrin and demonstrated that Arp1 binds directly to betaIII spectrin.</td>\n",
       "      <td>betaIII spectrin</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2025</th>\n",
       "      <td>In contrast to the model previously proposed ( 44), however, we found that paxillin alpha could bind directly to PAK3.</td>\n",
       "      <td>paxillin alpha</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2457</th>\n",
       "      <td>Git2 is highly homologous to p95PKL; and we indeed detected ternary complexes of paxillin α/Git2/βPIX, and Git2/βPIX/PAK3.</td>\n",
       "      <td>paxillin α</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2512</th>\n",
       "      <td>Overexpression in Chinese hamster ovary cells of an amphiphysin 1 fragment that binds both AP-2 and clathrin resulted in a segregation of clathrin, which acquired a diffuse distribution, from AP-2...</td>\n",
       "      <td>amphiphysin 1</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2523</th>\n",
       "      <td>The kinetics provide a detailed picture of the threefold difference in affinity observed for estriol binding to beta ( KD = 5.6 nM) and alpha ( KD = 18 nM) isoforms of the receptor.</td>\n",
       "      <td>alpha</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2524</th>\n",
       "      <td>The kinetics provide a detailed picture of the threefold difference in affinity observed for estriol binding to beta ( KD = 5.6 nM) and alpha ( KD = 18 nM) isoforms of the receptor.</td>\n",
       "      <td>beta</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2590</th>\n",
       "      <td>Although the authors claim that this effect is through disruption of PKC binding to RACK1, one of the residues converted to alanine was I236, a residue we now know to be essential for interaction ...</td>\n",
       "      <td>PDE4D5</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3022</th>\n",
       "      <td>Synaptotagmin binds beta-SNAP, but not alpha-SNAP, with a high degree of selectivity ( Schiavo et al. 1995 ); however, the physiological function of this interaction has yet to be established.</td>\n",
       "      <td>alpha-SNAP</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3024</th>\n",
       "      <td>Synaptotagmin binds beta-SNAP, but not alpha-SNAP, with a high degree of selectivity ( Schiavo et al. 1995 ); however, the physiological function of this interaction has yet to be established.</td>\n",
       "      <td>beta-SNAP</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3904</th>\n",
       "      <td>Treatment with a phospholipid known to decrease self-association of ERMs enhances the binding of merlin isoform 1 to NHE-RF.</td>\n",
       "      <td>merlin isoform 1</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5203</th>\n",
       "      <td>In addition, mutations of the two heparin binding sites in intact apoE3 demonstrated the dominant role of the N-terminal site in the heparin binding of apoE even in the lipid-free state.</td>\n",
       "      <td>apoE3</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5395</th>\n",
       "      <td>The PI bound to PI-TPalpha is delivered as a substrate for phospholipase A2 ( PLA2).</td>\n",
       "      <td>PI-TPalpha</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5417</th>\n",
       "      <td>This is consistent with the most recent results with myosin Iβ and myosin V and supports the idea that calmodulin dissociation is not the inhibitory mechanism of calmodulin binding myosins.</td>\n",
       "      <td>myosin Iβ</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5482</th>\n",
       "      <td>This indicates that DR4 binds to DM-DO with higher affinity and/or stability than to DM.</td>\n",
       "      <td>DR4</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                         text  \\\n",
       "9                                                        The yeast two-hybrid interaction assays indicated that filamin-A does not bind to any integrin β subunit, whereas filamin-B only interacts with β1A.   \n",
       "307   Experimental studies of liver regeneration, the acute-phase response to inflammation, and the pathology of myotonic dystrophy (DM) showed that, in all of these situations, CUGBP1 binds to the GCN ...   \n",
       "1447                                                                                                  A neurobeachin isoform that does not bind RII, beige-like protein (BGL), is expressed in many tissues.    \n",
       "1715  Dynamin 2, a binding partner of the SH3 domain of amphiphysin (9), was recruited to the tubules when coexpressed with M-amphiphysin 2 (Fig. 2D), but not when coexpressed with its BAR* domain, whic...   \n",
       "1991   Baker et al. (Baker et al., 2001 ) show in vitro experiments in which MAE binding to PNT-P2 leads to activation of transcription, and assume that this occurs via MAE promoting MAPK phosphorylatio...   \n",
       "2007                  To test this hypothesis, we have examined the interaction of Arp1 with the Golgi-associated betaIII isoform of spectrin and demonstrated that Arp1 binds directly to betaIII spectrin.    \n",
       "2008                  To test this hypothesis, we have examined the interaction of Arp1 with the Golgi-associated betaIII isoform of spectrin and demonstrated that Arp1 binds directly to betaIII spectrin.    \n",
       "2025                                                                                  In contrast to the model previously proposed ( 44), however, we found that paxillin alpha could bind directly to PAK3.    \n",
       "2457                                                                               Git2 is highly homologous to p95PKL; and we indeed detected ternary complexes of paxillin α/Git2/βPIX, and Git2/βPIX/PAK3.   \n",
       "2512  Overexpression in Chinese hamster ovary cells of an amphiphysin 1 fragment that binds both AP-2 and clathrin resulted in a segregation of clathrin, which acquired a diffuse distribution, from AP-2...   \n",
       "2523                    The kinetics provide a detailed picture of the threefold difference in affinity observed for estriol binding to beta ( KD = 5.6 nM) and alpha ( KD = 18 nM) isoforms of the receptor.   \n",
       "2524                    The kinetics provide a detailed picture of the threefold difference in affinity observed for estriol binding to beta ( KD = 5.6 nM) and alpha ( KD = 18 nM) isoforms of the receptor.   \n",
       "2590  Although the authors claim that this effect is through disruption of PKC binding to RACK1, one of the residues converted to alanine was I236, a residue we now know to be essential for interaction ...   \n",
       "3022         Synaptotagmin binds beta-SNAP, but not alpha-SNAP, with a high degree of selectivity ( Schiavo et al. 1995 ); however, the physiological function of this interaction has yet to be established.   \n",
       "3024         Synaptotagmin binds beta-SNAP, but not alpha-SNAP, with a high degree of selectivity ( Schiavo et al. 1995 ); however, the physiological function of this interaction has yet to be established.   \n",
       "3904                                                                             Treatment with a phospholipid known to decrease self-association of ERMs enhances the binding of merlin isoform 1 to NHE-RF.   \n",
       "5203               In addition, mutations of the two heparin binding sites in intact apoE3 demonstrated the dominant role of the N-terminal site in the heparin binding of apoE even in the lipid-free state.   \n",
       "5395                                                                                                                     The PI bound to PI-TPalpha is delivered as a substrate for phospholipase A2 ( PLA2).   \n",
       "5417            This is consistent with the most recent results with myosin Iβ and myosin V and supports the idea that calmodulin dissociation is not the inhibitory mechanism of calmodulin binding myosins.   \n",
       "5482                                                                                                                 This indicates that DR4 binds to DM-DO with higher affinity and/or stability than to DM.   \n",
       "\n",
       "                          name            label  length  \n",
       "9                          β1A  protein-isoform       3  \n",
       "307                        LIP  protein-isoform       3  \n",
       "1447      neurobeachin isoform  protein-isoform      20  \n",
       "1715           M-amphiphysin 2  protein-isoform      15  \n",
       "1991                    PNT-P2  protein-isoform       6  \n",
       "2007  Golgi-associated betaIII  protein-isoform      24  \n",
       "2008          betaIII spectrin  protein-isoform      16  \n",
       "2025            paxillin alpha  protein-isoform      14  \n",
       "2457                paxillin α  protein-isoform      10  \n",
       "2512             amphiphysin 1  protein-isoform      13  \n",
       "2523                     alpha  protein-isoform       5  \n",
       "2524                      beta  protein-isoform       4  \n",
       "2590                    PDE4D5  protein-isoform       6  \n",
       "3022                alpha-SNAP  protein-isoform      10  \n",
       "3024                 beta-SNAP  protein-isoform       9  \n",
       "3904          merlin isoform 1  protein-isoform      16  \n",
       "5203                     apoE3  protein-isoform       5  \n",
       "5395                PI-TPalpha  protein-isoform      10  \n",
       "5417                 myosin Iβ  protein-isoform       9  \n",
       "5482                       DR4  protein-isoform       3  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_train[df_train[\"label\"] == \"protein-isoform\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>This is in agreement with the binding affinity of native apoE isoforms for Abeta (apoE2 &gt; apoE3 &gt; apoE4) ( 15, 41) and reduced LRP-dependent uptake of apoE4/Abeta complexes in CHO cells ( 38).</td>\n",
       "      <td>apoE2</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>This is in agreement with the binding affinity of native apoE isoforms for Abeta (apoE2 &gt; apoE3 &gt; apoE4) ( 15, 41) and reduced LRP-dependent uptake of apoE4/Abeta complexes in CHO cells ( 38).</td>\n",
       "      <td>apoE3</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>This is in agreement with the binding affinity of native apoE isoforms for Abeta (apoE2 &gt; apoE3 &gt; apoE4) ( 15, 41) and reduced LRP-dependent uptake of apoE4/Abeta complexes in CHO cells ( 38).</td>\n",
       "      <td>apoE4</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>Instead, we find that nNOS binds to the second PDZ repeat of PSD-95, and a related protein PSD-93.</td>\n",
       "      <td>nNOS</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>The anti-FGF Ab showed that FGF-2 was bound to a similar extent to CD44v3-v10 and CD44vRA, whereas CD44s did not bind FGF-2.</td>\n",
       "      <td>CD44v3-v10</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>450</th>\n",
       "      <td>The anti-FGF Ab showed that FGF-2 was bound to a similar extent to CD44v3-v10 and CD44vRA, whereas CD44s did not bind FGF-2.</td>\n",
       "      <td>CD44vRA</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>694</th>\n",
       "      <td>These results suggest that l-afadin and vinculin bind to ponsin in a competitive manner, and that l-afadin, ponsin, and vinculin hardly form a ternary complex.</td>\n",
       "      <td>l-afadin</td>\n",
       "      <td>protein-isoform</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                 text  \\\n",
       "38   This is in agreement with the binding affinity of native apoE isoforms for Abeta (apoE2 > apoE3 > apoE4) ( 15, 41) and reduced LRP-dependent uptake of apoE4/Abeta complexes in CHO cells ( 38).   \n",
       "39   This is in agreement with the binding affinity of native apoE isoforms for Abeta (apoE2 > apoE3 > apoE4) ( 15, 41) and reduced LRP-dependent uptake of apoE4/Abeta complexes in CHO cells ( 38).   \n",
       "40   This is in agreement with the binding affinity of native apoE isoforms for Abeta (apoE2 > apoE3 > apoE4) ( 15, 41) and reduced LRP-dependent uptake of apoE4/Abeta complexes in CHO cells ( 38).   \n",
       "338                                                                                                Instead, we find that nNOS binds to the second PDZ repeat of PSD-95, and a related protein PSD-93.   \n",
       "449                                                                      The anti-FGF Ab showed that FGF-2 was bound to a similar extent to CD44v3-v10 and CD44vRA, whereas CD44s did not bind FGF-2.   \n",
       "450                                                                      The anti-FGF Ab showed that FGF-2 was bound to a similar extent to CD44v3-v10 and CD44vRA, whereas CD44s did not bind FGF-2.   \n",
       "694                                   These results suggest that l-afadin and vinculin bind to ponsin in a competitive manner, and that l-afadin, ponsin, and vinculin hardly form a ternary complex.   \n",
       "\n",
       "           name            label  length  \n",
       "38        apoE2  protein-isoform       5  \n",
       "39        apoE3  protein-isoform       5  \n",
       "40        apoE4  protein-isoform       5  \n",
       "338        nNOS  protein-isoform       4  \n",
       "449  CD44v3-v10  protein-isoform      10  \n",
       "450     CD44vRA  protein-isoform       7  \n",
       "694    l-afadin  protein-isoform       8  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_dev[df_dev[\"label\"] == \"protein-isoform\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Extracts were incubated with 20 mug of Gst-RalGDS-RBD precoupled to gluthathione beads to isolate GTP-bound (activated) Rap1.</td>\n",
       "      <td>Gst-RalGDS-RBD</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>Although studies using the fusion protein CTLA4-Ig, which binds to B7.1 and B7.2, have shown suppression of cell-mediated and humoral immunity in several mouse models of autoimmune disease (12–16)...</td>\n",
       "      <td>CTLA4-Ig</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>Thus neomycin, polylysine, and spermine reversed the antagonism of TNP-ATP binding to MBP_1.1C by PIP2.</td>\n",
       "      <td>MBP_1.1C</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>758</th>\n",
       "      <td>Assay for CD22Rg Binding to HEC After washing with Medium 199, HEC were incubated with 2.5 mug/ ml CD22 mRg in Medium 199, 1% BSA at 4 degrees C for 2 h, in the absence or presence of alpha2-6-sLa...</td>\n",
       "      <td>CD22Rg</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949</th>\n",
       "      <td>As expected, conversion of the TAGA element into a consensus TATA box increased f:TFIID binding to the LRP-2 promoter, leading to increased protection of the TATA box region and downstream promote...</td>\n",
       "      <td>f:TFIID</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>975</th>\n",
       "      <td>The relative proportion of TBP bound to GST-TADIV of the total present in the crude extract (Fig. 2) was found to be about 1/10 of that of the TAFIIs as determined by densitometric analysis (data ...</td>\n",
       "      <td>GST-TADIV</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1083</th>\n",
       "      <td>Interestingly, dTAFII110 also bound GST-IIAτ more abundantly than it bound to GST-αβ (Fig.4 C, lane 4).</td>\n",
       "      <td>GST-αβ</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1084</th>\n",
       "      <td>Interestingly, dTAFII110 also bound GST-IIAτ more abundantly than it bound to GST-αβ (Fig.4 C, lane 4).</td>\n",
       "      <td>GST-IIAτ</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2852</th>\n",
       "      <td>Whereas the binding of ATP had no effect on the binding of IkappaBalpha peptide for the rhIKK-1/rhIKK-2 heterodimer, the binding of ATP decreased the affinity of IkappaBalpha peptide for rhIKK-i a...</td>\n",
       "      <td>rhIKK-i</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2853</th>\n",
       "      <td>Whereas the binding of ATP had no effect on the binding of IkappaBalpha peptide for the rhIKK-1/rhIKK-2 heterodimer, the binding of ATP decreased the affinity of IkappaBalpha peptide for rhIKK-i a...</td>\n",
       "      <td>rhTBK-1</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2854</th>\n",
       "      <td>Whereas the binding of ATP had no effect on the binding of IkappaBalpha peptide for the rhIKK-1/rhIKK-2 heterodimer, the binding of ATP decreased the affinity of IkappaBalpha peptide for rhIKK-i a...</td>\n",
       "      <td>rhIKK-1</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2858</th>\n",
       "      <td>Whereas the binding of ATP had no effect on the binding of IkappaBalpha peptide for the rhIKK-1/rhIKK-2 heterodimer, the binding of ATP decreased the affinity of IkappaBalpha peptide for rhIKK-i a...</td>\n",
       "      <td>rhIKK-2</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3040</th>\n",
       "      <td>To identify which region of HP1 binds to the H4 peptides, we used an rHP1 sample in which the protein had been cleaved into two peptides.</td>\n",
       "      <td>rHP1</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3218</th>\n",
       "      <td>From the earliest demonstrations that CRP binds to microbes and mediates their killing in vitro, to more recent observations of protection against Gram-positive and Gram-negative bacteria in CRPtg...</td>\n",
       "      <td>CRPtg</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3717</th>\n",
       "      <td>Binding of Dfd and Dfd-VP16 to the 2 ϫ D site (see Figure 4 for sequence) in EMSAs.</td>\n",
       "      <td>Dfd-VP16</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4051</th>\n",
       "      <td>The interaction between synthetic peptides and AtCaM isoforms was analyzed by applying of synthetic peptides to a nitrocellulose membrane in a slot blot apparatus, and incubating the membrane with...</td>\n",
       "      <td>35S-labeled AtCaM</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4124</th>\n",
       "      <td>GST alone did not bind calmodulin in the presence of Ca2+ or EGTA and neither GST-dUNC13 or any other fusion protein described below bound to agarose beads under any Ca2+ conditions (data not show...</td>\n",
       "      <td>GST-dUNC13</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4275</th>\n",
       "      <td>To demonstrate that the reduction in the amount of Axin bound to Xgsk-3 is specifically dependent on GBP binding to Xgsk-3, the coimmunoprecipitation experiment was repeated using a high dose of H...</td>\n",
       "      <td>GBPwt</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4276</th>\n",
       "      <td>To demonstrate that the reduction in the amount of Axin bound to Xgsk-3 is specifically dependent on GBP binding to Xgsk-3, the coimmunoprecipitation experiment was repeated using a high dose of H...</td>\n",
       "      <td>HA-tagged GBP</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4278</th>\n",
       "      <td>To demonstrate that the reduction in the amount of Axin bound to Xgsk-3 is specifically dependent on GBP binding to Xgsk-3, the coimmunoprecipitation experiment was repeated using a high dose of H...</td>\n",
       "      <td>GBPmut</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4279</th>\n",
       "      <td>To demonstrate that the reduction in the amount of Axin bound to Xgsk-3 is specifically dependent on GBP binding to Xgsk-3, the coimmunoprecipitation experiment was repeated using a high dose of H...</td>\n",
       "      <td>HA-tagged GBP mutant</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4770</th>\n",
       "      <td>CTD binding and allosteric activation of the mammalian guanylyltransferase Mce1 was studied by introducing alanine substitutions for Tyr1, Pro3, Thr4, or Pro6 of each heptad of the 28-mer syntheti...</td>\n",
       "      <td>CTD Ser5-PO4</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4886</th>\n",
       "      <td>The actin binding of A-fascin was the same as that of unphosphorylated r-fascin regardless of the treatment of A-fascin with PKC, indicating that Ser-39 is responsible for the phosphorylation-depe...</td>\n",
       "      <td>r-fascin</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4914</th>\n",
       "      <td>Tyr177 lies within a Grb2 binding motif (Y-X-N-X) in BCR, and is essential for Grb2 binding to BCR/ABL (Pendergast et al., 1993 ; Puil et al., 1994 ).</td>\n",
       "      <td>BCR/ABL</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                         text  \\\n",
       "41                                                                              Extracts were incubated with 20 mug of Gst-RalGDS-RBD precoupled to gluthathione beads to isolate GTP-bound (activated) Rap1.   \n",
       "191   Although studies using the fusion protein CTLA4-Ig, which binds to B7.1 and B7.2, have shown suppression of cell-mediated and humoral immunity in several mouse models of autoimmune disease (12–16)...   \n",
       "205                                                                                                   Thus neomycin, polylysine, and spermine reversed the antagonism of TNP-ATP binding to MBP_1.1C by PIP2.   \n",
       "758   Assay for CD22Rg Binding to HEC After washing with Medium 199, HEC were incubated with 2.5 mug/ ml CD22 mRg in Medium 199, 1% BSA at 4 degrees C for 2 h, in the absence or presence of alpha2-6-sLa...   \n",
       "949   As expected, conversion of the TAGA element into a consensus TATA box increased f:TFIID binding to the LRP-2 promoter, leading to increased protection of the TATA box region and downstream promote...   \n",
       "975   The relative proportion of TBP bound to GST-TADIV of the total present in the crude extract (Fig. 2) was found to be about 1/10 of that of the TAFIIs as determined by densitometric analysis (data ...   \n",
       "1083                                                                                                  Interestingly, dTAFII110 also bound GST-IIAτ more abundantly than it bound to GST-αβ (Fig.4 C, lane 4).   \n",
       "1084                                                                                                  Interestingly, dTAFII110 also bound GST-IIAτ more abundantly than it bound to GST-αβ (Fig.4 C, lane 4).   \n",
       "2852  Whereas the binding of ATP had no effect on the binding of IkappaBalpha peptide for the rhIKK-1/rhIKK-2 heterodimer, the binding of ATP decreased the affinity of IkappaBalpha peptide for rhIKK-i a...   \n",
       "2853  Whereas the binding of ATP had no effect on the binding of IkappaBalpha peptide for the rhIKK-1/rhIKK-2 heterodimer, the binding of ATP decreased the affinity of IkappaBalpha peptide for rhIKK-i a...   \n",
       "2854  Whereas the binding of ATP had no effect on the binding of IkappaBalpha peptide for the rhIKK-1/rhIKK-2 heterodimer, the binding of ATP decreased the affinity of IkappaBalpha peptide for rhIKK-i a...   \n",
       "2858  Whereas the binding of ATP had no effect on the binding of IkappaBalpha peptide for the rhIKK-1/rhIKK-2 heterodimer, the binding of ATP decreased the affinity of IkappaBalpha peptide for rhIKK-i a...   \n",
       "3040                                                                To identify which region of HP1 binds to the H4 peptides, we used an rHP1 sample in which the protein had been cleaved into two peptides.   \n",
       "3218  From the earliest demonstrations that CRP binds to microbes and mediates their killing in vitro, to more recent observations of protection against Gram-positive and Gram-negative bacteria in CRPtg...   \n",
       "3717                                                                                                                      Binding of Dfd and Dfd-VP16 to the 2 ϫ D site (see Figure 4 for sequence) in EMSAs.   \n",
       "4051  The interaction between synthetic peptides and AtCaM isoforms was analyzed by applying of synthetic peptides to a nitrocellulose membrane in a slot blot apparatus, and incubating the membrane with...   \n",
       "4124  GST alone did not bind calmodulin in the presence of Ca2+ or EGTA and neither GST-dUNC13 or any other fusion protein described below bound to agarose beads under any Ca2+ conditions (data not show...   \n",
       "4275  To demonstrate that the reduction in the amount of Axin bound to Xgsk-3 is specifically dependent on GBP binding to Xgsk-3, the coimmunoprecipitation experiment was repeated using a high dose of H...   \n",
       "4276  To demonstrate that the reduction in the amount of Axin bound to Xgsk-3 is specifically dependent on GBP binding to Xgsk-3, the coimmunoprecipitation experiment was repeated using a high dose of H...   \n",
       "4278  To demonstrate that the reduction in the amount of Axin bound to Xgsk-3 is specifically dependent on GBP binding to Xgsk-3, the coimmunoprecipitation experiment was repeated using a high dose of H...   \n",
       "4279  To demonstrate that the reduction in the amount of Axin bound to Xgsk-3 is specifically dependent on GBP binding to Xgsk-3, the coimmunoprecipitation experiment was repeated using a high dose of H...   \n",
       "4770  CTD binding and allosteric activation of the mammalian guanylyltransferase Mce1 was studied by introducing alanine substitutions for Tyr1, Pro3, Thr4, or Pro6 of each heptad of the 28-mer syntheti...   \n",
       "4886  The actin binding of A-fascin was the same as that of unphosphorylated r-fascin regardless of the treatment of A-fascin with PKC, indicating that Ser-39 is responsible for the phosphorylation-depe...   \n",
       "4914                                                   Tyr177 lies within a Grb2 binding motif (Y-X-N-X) in BCR, and is essential for Grb2 binding to BCR/ABL (Pendergast et al., 1993 ; Puil et al., 1994 ).   \n",
       "\n",
       "                      name           label  length  \n",
       "41          Gst-RalGDS-RBD  fusion-protein      14  \n",
       "191               CTLA4-Ig  fusion-protein       8  \n",
       "205               MBP_1.1C  fusion-protein       8  \n",
       "758                 CD22Rg  fusion-protein       6  \n",
       "949                f:TFIID  fusion-protein       7  \n",
       "975              GST-TADIV  fusion-protein       9  \n",
       "1083                GST-αβ  fusion-protein       6  \n",
       "1084              GST-IIAτ  fusion-protein       8  \n",
       "2852               rhIKK-i  fusion-protein       7  \n",
       "2853               rhTBK-1  fusion-protein       7  \n",
       "2854               rhIKK-1  fusion-protein       7  \n",
       "2858               rhIKK-2  fusion-protein       7  \n",
       "3040                  rHP1  fusion-protein       4  \n",
       "3218                 CRPtg  fusion-protein       5  \n",
       "3717              Dfd-VP16  fusion-protein       8  \n",
       "4051     35S-labeled AtCaM  fusion-protein      17  \n",
       "4124            GST-dUNC13  fusion-protein      10  \n",
       "4275                 GBPwt  fusion-protein       5  \n",
       "4276         HA-tagged GBP  fusion-protein      13  \n",
       "4278                GBPmut  fusion-protein       6  \n",
       "4279  HA-tagged GBP mutant  fusion-protein      20  \n",
       "4770          CTD Ser5-PO4  fusion-protein      12  \n",
       "4886              r-fascin  fusion-protein       8  \n",
       "4914               BCR/ABL  fusion-protein       7  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_train[df_train[\"label\"] == \"fusion-protein\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>name</th>\n",
       "      <th>label</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>We found that the GST-SET domain fusion protein selectively retained histones H3 and H4, whereas binding to H2A and H2B was significantly weaker.</td>\n",
       "      <td>GST-SET domain fusion protein</td>\n",
       "      <td>fusion-protein</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                  text  \\\n",
       "583  We found that the GST-SET domain fusion protein selectively retained histones H3 and H4, whereas binding to H2A and H2B was significantly weaker.   \n",
       "\n",
       "                              name           label  length  \n",
       "583  GST-SET domain fusion protein  fusion-protein      29  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_dev[df_dev[\"label\"] == \"fusion-protein\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Entities grouped by their length"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>label</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DNA</th>\n",
       "      <td>9.201102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RNA</th>\n",
       "      <td>8.863636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RNA-family</th>\n",
       "      <td>5.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>amino-acid</th>\n",
       "      <td>6.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assay</th>\n",
       "      <td>10.789474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>brand</th>\n",
       "      <td>8.750000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell</th>\n",
       "      <td>8.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chemical</th>\n",
       "      <td>6.107969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>disease</th>\n",
       "      <td>5.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>drug</th>\n",
       "      <td>8.272727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>experiment-tag</th>\n",
       "      <td>5.383838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>experimental-construct</th>\n",
       "      <td>12.384615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fusion-protein</th>\n",
       "      <td>8.708333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <td>8.316327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mutation</th>\n",
       "      <td>6.461538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>organelle</th>\n",
       "      <td>8.900000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>organism</th>\n",
       "      <td>13.750000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>parameter</th>\n",
       "      <td>2.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>peptide</th>\n",
       "      <td>10.884615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>process</th>\n",
       "      <td>17.100000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein</th>\n",
       "      <td>6.092871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-DNA-complex</th>\n",
       "      <td>13.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-RNA-complex</th>\n",
       "      <td>7.928571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-complex</th>\n",
       "      <td>8.768977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-domain</th>\n",
       "      <td>8.948936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-family</th>\n",
       "      <td>7.367021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-isoform</th>\n",
       "      <td>10.050000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-motif</th>\n",
       "      <td>9.204082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-region</th>\n",
       "      <td>10.692308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>reagent</th>\n",
       "      <td>12.369565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>5.500000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           length\n",
       "label                            \n",
       "DNA                      9.201102\n",
       "RNA                      8.863636\n",
       "RNA-family               5.500000\n",
       "amino-acid               6.333333\n",
       "assay                   10.789474\n",
       "brand                    8.750000\n",
       "cell                     8.000000\n",
       "chemical                 6.107969\n",
       "disease                  5.000000\n",
       "drug                     8.272727\n",
       "experiment-tag           5.383838\n",
       "experimental-construct  12.384615\n",
       "fusion-protein           8.708333\n",
       "gene                     8.316327\n",
       "mutation                 6.461538\n",
       "organelle                8.900000\n",
       "organism                13.750000\n",
       "parameter                2.500000\n",
       "peptide                 10.884615\n",
       "process                 17.100000\n",
       "protein                  6.092871\n",
       "protein-DNA-complex     13.333333\n",
       "protein-RNA-complex      7.928571\n",
       "protein-complex          8.768977\n",
       "protein-domain           8.948936\n",
       "protein-family           7.367021\n",
       "protein-isoform         10.050000\n",
       "protein-motif            9.204082\n",
       "protein-region          10.692308\n",
       "reagent                 12.369565\n",
       "tissue                   5.500000"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_train.groupby(\"label\").mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>label</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DNA</th>\n",
       "      <td>9.674419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RNA</th>\n",
       "      <td>6.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>amino-acid</th>\n",
       "      <td>6.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assay</th>\n",
       "      <td>6.222222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell</th>\n",
       "      <td>7.222222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chemical</th>\n",
       "      <td>6.133333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>disease</th>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>drug</th>\n",
       "      <td>9.600000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>experiment-tag</th>\n",
       "      <td>3.615385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>experimental-construct</th>\n",
       "      <td>10.600000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fusion-protein</th>\n",
       "      <td>29.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <td>4.750000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>organelle</th>\n",
       "      <td>10.750000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>organism</th>\n",
       "      <td>7.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>peptide</th>\n",
       "      <td>10.125000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>process</th>\n",
       "      <td>12.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein</th>\n",
       "      <td>6.224138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-RNA-complex</th>\n",
       "      <td>6.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-complex</th>\n",
       "      <td>10.489362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-domain</th>\n",
       "      <td>6.205882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-family</th>\n",
       "      <td>6.822034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-isoform</th>\n",
       "      <td>6.285714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-motif</th>\n",
       "      <td>6.636364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein-region</th>\n",
       "      <td>8.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>reagent</th>\n",
       "      <td>12.285714</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           length\n",
       "label                            \n",
       "DNA                      9.674419\n",
       "RNA                      6.333333\n",
       "amino-acid               6.333333\n",
       "assay                    6.222222\n",
       "cell                     7.222222\n",
       "chemical                 6.133333\n",
       "disease                  2.000000\n",
       "drug                     9.600000\n",
       "experiment-tag           3.615385\n",
       "experimental-construct  10.600000\n",
       "fusion-protein          29.000000\n",
       "gene                     4.750000\n",
       "organelle               10.750000\n",
       "organism                 7.000000\n",
       "peptide                 10.125000\n",
       "process                 12.666667\n",
       "protein                  6.224138\n",
       "protein-RNA-complex      6.000000\n",
       "protein-complex         10.489362\n",
       "protein-domain           6.205882\n",
       "protein-family           6.822034\n",
       "protein-isoform          6.285714\n",
       "protein-motif            6.636364\n",
       "protein-region           8.500000\n",
       "reagent                 12.285714"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_dev.groupby(\"label\").mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "def checkDuplicateEntities(path):\n",
    "    \n",
    "    with open(path, \"r\", encoding=\"utf-8\") as file:\n",
    "        data = json.load(file)\n",
    "\n",
    "    cnt=0\n",
    "    TEXT = []\n",
    "    ENTITY_NAME = []\n",
    "    ENTITY_LABEL = []\n",
    "    IDS = []\n",
    "#     IDS = []\n",
    "    for i, dataItem in enumerate(data):\n",
    "        entitiesDict = {}\n",
    "        entitiesDict[\"entities\"] = []\n",
    "        TEXT.append(dataItem[\"text\"])\n",
    "        if dataItem[\"id\"] in IDS:\n",
    "            IDS.append(dataItem[\"id\"])\n",
    "            print(dataItem[\"id\"], dataItem[\"text\"])\n",
    "            cnt+=1\n",
    "        else:\n",
    "            IDS.append(dataItem[\"id\"])\n",
    "            \n",
    "    return pd.DataFrame({\"text\": TEXT, \"id\": IDS})\n",
    "    \n",
    "    print(cnt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5083349 We next determined the regions of Mirk and HNF1α necessary for their interaction.\n",
      "3737303 We then examined the direct DNA binding of Smad5 and Smad4 to the GC-rich sequence of PBE.\n",
      "9486072 To identify the region within the cortactin amino terminus responsible for associating with the Arp2/3 complex, Sepharose beads covalently coupled to the NTA and repeats regions were incubated with 10T1/2 fibroblast lysate, and Arp2/3 binding was assayed by immunoblotting (Fig. 6 A). \n",
      "5301491 Menin Binds GTP\n",
      "19640896 Hes6 has been shown to act indirectly by binding other orange domain-containing proteins, rather than by acting as a DNA-binding transcription factor (Bae et al., 2000; Koyano-Nakagawa et al., 2000).\n",
      "4211455 Experiments using equilibrium dialysis (39) and protein fluorescence quenching (11) showed that NADH binds to isolated R. rubrum dI protein with a K d≈ 20 μM. \n",
      "4638492 Our immunotube technique revealed that A18 hnRNP binds to a large number of stress-responsive transcripts (Table I).\n",
      "11491303 The polypeptide encoded by the cDNA isolated above was expressed in Escherichia coli either as a His-tagged protein (His6-cl.50) or as a GST fusion protein (GST-cl.50). \n",
      "4470330 To evaluate the dynamics of the interaction between Hsp70 and an ARE, off-rate analyses were performed to measure the stability of Hsp70·Fl-TNFα ARE complexes in solution.\n",
      "3791724 To map further the regions required for binding of X11α with CCS, we used GST fusion protein binding assays.\n",
      "6619022 The proline-rich region of Cbl contains many possible binding sites for the SH3 domains of various proteins (19).\n",
      "12698116 Hypophosphorylated 4E-BP1 interacts strongly with eIF4E, whereas hyperphosphorylation of 4E-BP1 drastically decreases its binding to eIF4E.\n",
      "1689218 To determine whether the interaction of Pbx and Meis1 families extends beyond mammalian isoforms, the Pbx homologs from Drosophila and C. elegans were tested for cooperative CRS1 binding with Meis1.\n",
      "9490708 Most PDZ-containing proteins bind to their targets via a COOH-terminal sequence that conforms to an S/T-X-V/I motif or that consists of aromatic and/or hydrophobic residues (Kim et al. 1995; Kornau et al. 1995; Songyang et al. 1997; Daniels et al. 1998).\n",
      "4574093 To extend the binding investigation to dSPRY and dCbl, various constructs as depicted in Fig. 8 D were made.\n",
      "9328714 We have shown previously, using an in vitro peptide competition assay, that the peptide FGEYRSLESDNEE blocks binding of a biotin-labeled TGN38 peptide to AP-2 purified from rat brain (Kamiguchi et al., 1998b).\n",
      "2028450 To study the ATP binding properties of synapsins, we expressed their C-domains as GST fusion proteins and used 35S-labeled ATPγS as a non-hydrolyzable ATP ligand.\n",
      "11752077 The binding patterns of SIN3 and RPD3 on polytene chromosomes were highly coincident, suggesting that the SIN3-RPD3 complex is the major form of chromatin-bound SIN3 and RPD3 in salivary gland cells.\n",
      "2692783 To examine whether Axin binds directly to plakoglobin, GST-plakoglobin (full length) and MBP-rAxin (full length) were purified (Fig.2 B).\n"
     ]
    }
   ],
   "source": [
    "PATH = \"/home/abbhatt/Documents/flairExperiments/data/train-refactored.json\"\n",
    "df_check_train = checkDuplicateEntities(PATH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>793</th>\n",
       "      <td>Binding of the 4E-BPs to eIF4E is reversible and is dependent on the phosphorylation status of 4E-BP.</td>\n",
       "      <td>12698116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>794</th>\n",
       "      <td>Hypophosphorylated 4E-BP1 interacts strongly with eIF4E, whereas hyperphosphorylation of 4E-BP1 drastically decreases its binding to eIF4E.</td>\n",
       "      <td>12698116</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                            text  \\\n",
       "793                                       Binding of the 4E-BPs to eIF4E is reversible and is dependent on the phosphorylation status of 4E-BP.    \n",
       "794  Hypophosphorylated 4E-BP1 interacts strongly with eIF4E, whereas hyperphosphorylation of 4E-BP1 drastically decreases its binding to eIF4E.   \n",
       "\n",
       "           id  \n",
       "793  12698116  \n",
       "794  12698116  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_check_train[df_check_train[\"id\"] == 12698116]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1746313 Interaction between the repeated motifs of EPRS and IRS was previously shown by genetic analysis (18).\n",
      "6619094 GST-Cbl was precipitated, and the binding of Src and Abl was analyzed by Western blotting using anti-Src and anti-Abl antibodies (Fig. ​44A). \n",
      "3156714 The studies described above demonstrate that annexin II mediates the binding of β2GPI to endothelial cells but do not address the issue of whether phospholipid, glycosaminoglycans, or other cell surface proteins are also required for this interaction. \n"
     ]
    }
   ],
   "source": [
    "PATH = \"/home/abbhatt/Documents/flairExperiments/data/dev-refactored.json\"\n",
    "df_check_dev = checkDuplicateEntities(PATH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>Measurement of the Binding of Annexin II to β2GPI by Surface Plasmon Resonance</td>\n",
       "      <td>3156714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>The studies described above demonstrate that annexin II mediates the binding of β2GPI to endothelial cells but do not address the issue of whether phospholipid, glycosaminoglycans, or other cell s...</td>\n",
       "      <td>3156714</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                        text  \\\n",
       "160                                                                                                                           Measurement of the Binding of Annexin II to β2GPI by Surface Plasmon Resonance   \n",
       "161  The studies described above demonstrate that annexin II mediates the binding of β2GPI to endothelial cells but do not address the issue of whether phospholipid, glycosaminoglycans, or other cell s...   \n",
       "\n",
       "          id  \n",
       "160  3156714  \n",
       "161  3156714  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_check_dev[df_check_dev[\"id\"] == 3156714]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1405, 2)"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_check_train.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(201, 2)"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_check_dev.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
